### ISSUE TITLE
Guidelines for code

### ISSUE THREAD

**Issue by shymega** [debauchee/barrier#716]
We should make a list of guidelines for code contributions.

For example, with a review comment in #703, it is pointed out about tabs and spaces in the contribution. We could use [EditorConfig](https://editorconfig.org/) to assist with these guidelines to an extent, depending on the editor used by contributor(s).

I'm assigning this issue to myself to create a configuration file if the suggestion gets a positive reaction.

**Comment by p12tic** [comment]
I think it's enough if we had the guidelines themselves somewhere. People can then configure their editors according to them. For example, EditorConfig does not support my editor of choice, I guess there will more people with similar issues :)

**Comment by shymega** [comment]
Sure, we can have the guidelines on the wiki/repo (or both), but there's nothing wrong with having EditorConfig available as well. Just because it doesn't support X editor doesn't make it inherently unnecessary. What editor do you use incidentally?

**Comment by p12tic** [comment]
Agreed. I use QtCreator.

**Comment by shymega** [comment]
Ah, I've never used that. I use VS Code. I'll work on a basic EditorConfig for now, and create a document in the repo's root. Thoughts on creating a Wiki page for it? We could do with a Contributing page, so perhaps integrate into that?

**Comment by shymega** [comment]
@p12tic Is 4 spaces alright? EditorConfig also allows specifying of line endings.. so we could stick to CR or CRLF.

**Comment by p12tic** [comment]
Yes, 4 spaces is good. For line endings we should use LF as anything other will introduce problems. Git recomments using LF for cross-platform projects.

**Comment by shymega** [comment]
EditorConfig committed. Wiki page is the next step.
